### DISCUSSION TITLE
GK Flags - Install Fails

### DISCUSSION THREAD

**Discussion by michallawson** [HCL-TECH-SOFTWARE/connections-automation#103] [Q&A]
My installation runs through until it gets to point where it updates the GK flags, when I get teh error:

ASK [component-pack : Enable the Community GK flags for comm-highlight] **************************************************************************************************************************************
fatal: [cmzr.campbellreith.net]: FAILED! => {"access_control_allow_headers": "x-requested-with, Content-Type, origin, authorization, accept, client-security-token", "access_control_allow_methods": "POST, GET, OPTIONS, DELETE, PUT", "access_control_max_age": "1000", "changed": false, "connection": "close", "content": "<!DOCTYPE HTML PUBLIC \"-//IETF//DTD HTML 2.0//EN\">\n<html><head>\n<title>500 Internal Server Error</title>\n</head><body>\n<h1>Internal Server Error</h1>\n<p>The server encountered an internal error or\nmisconfiguration and was unable to complete\nyour request.</p>\n<p>Please contact the server administrator,\n you@your.address and inform them of the time the error occurred,\nand anything you might have done that may have\ncaused the error.</p>\n<p>More information about this error may be available\nin the server error log.</p>\n<hr>\n<address>IBM_HTTP_Server at connect.campbellreith.net Port 443</address>\n</body></html>\n", "content_length": "614", "content_type": "text/html; charset=iso-8859-1", "date": "Wed, 07 Apr 2021 02:32:36 GMT", "elapsed": 0, "msg": "Status code was 500 and not [200]: HTTP Error 500: Internal Server Error", "redirected": false, "server": "nginx/1.16.1", "status": 500, "url": "https://connect.campbellreith.net/connections/config/basic/rest/gatekeeper/", "vary": "Accept-Encoding"}

If I then access https://connect.campbellreith.net/connections/config/basic/rest/gatekeeper through Firefox I receive the following message.
	
title	"Error received"
status	"500"
message	"Unauthenticated access is not allowed"

I can logon to connections via https://connect.campbellreith.net/homepage and while logged in if I then run https://connect.campbellreith.net/connections/config/basic/rest/gatekeeper again I get:

organization	"00000000-0000-0000-0000-000000000000"
role		"admin"
settings	[…]
source		"highway"

I am using the small deployment and the quickstart environment with:

connections.example.com => cnx07.campbellreith.net
web.example.com => connect.campbellreith.net

Will be grateful if you can point me in the direction of where I should look to resolve the problem.
Kind regards
Michael Lawson

as follows :

and quick start

**Comment by dejo1307** [comment]
Hey Michael,

I've just checked the code - it is using was_username and was_password credentials to log in to it. Can you check if in your component pack inventory file those two values are the same as those in connections inventory file (since those in connections inventory file are/were used when the CNX side was set up). I guess it is trying to log with those and getting 500 and that's why it fails to authenticate. 

On that topic - if that is blocking you, that step can be skipped since it is only enabling new wizard (can be enabled manually going to config as well). On another note, we are working in the last two weeks on a quite big refactoring to get rid of those issues and to both centralise the variables in a single groups file and to get less chance to mess something in one file but not in the other (deduplication). 

Please let me know how that goes

Best,
Dejan

**Reply by michallawson** [reply]
Dejan,
Thank you.

I have checked and it is using the same username and password, which also works when logging via Firefox. Will skip the GK flags section and try again, otherwise will wait for the update you are working on.

Regards
Michael

**Reply by michallawson** [reply]
Dejan,

Have been a little reading and docs.ansible.com it states that the validate_certs  boolean was added in 1.9.2 of ansible and the choices are no or yes. In the task for GK Flags validate_certs is stated to be false. Do you think it could be this which is causing the problem?

Seems the default could have changed from no prior to 1.9.2 to yes post 1.9.2

Regards

Michael

**Reply by dejo1307** [reply]
Hey Michael, do you mind just testing it locally and letting me know if it works for you? I am having hard time reproducing it. I got 500 once today, but figured out it had to do with me running wrong playbooks and CNX never being started.

**Reply by michallawson** [reply]
Changing validate_certs to "no" did the trick as it ran through for me when I tested it.

Installation completed successfully, but the only trouble I now have is chrome. All works with Firefox, but on chrome I get a cross site cookie error and will not allow me to logon.

**Reply by dejo1307** [reply]
Thanks a lot! Regarding the last issue, run that playbooks/hcl/connections-post-install.yml with your connections inventory file, that should fix it.

**Comment by michallawson** [comment]
Dejan

Thank you for your help and seems to be working.

Will do some tests to make sure everything is okay, but looks good so far.
Kind regards

Michael

**Reply by dejo1307** [reply]
You're welcome Michael, happy to help like always :)

Best regards,
Dejan
